test/unit/transform/ferc1_test.py

```diff
@@ -548,6 +556,207 @@ def test_calculate_values_from_components():
     )
 
 
+def test_many_to_many_subdimensions_calculate_values_from_components():
```

Comment:
there are effectively 4 different unit tests in here all using the same lil helper `get_reconcilded_data` to build and poop out the corrected calculations. This could easily be broken up into 4 unit tests. i kinda think it is easier to read all in one place bc it is 4 iterations on the same idea. but very open to rejiggering.